deps/llvm.mk

```diff
@@ -66,7 +66,7 @@ LLVM_CXXFLAGS += $(CXXFLAGS)
 LLVM_CPPFLAGS += $(CPPFLAGS)
 LLVM_LDFLAGS += $(LDFLAGS)
 LLVM_CMAKE += -DLLVM_TARGETS_TO_BUILD:STRING="$(LLVM_TARGETS)" -DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD="$(LLVM_EXPERIMENTAL_TARGETS)" -DCMAKE_BUILD_TYPE="$(LLVM_CMAKE_BUILDTYPE)"
-LLVM_CMAKE += -DLLVM_ENABLE_ZLIB=OFF -DLLVM_ENABLE_LIBXML2=OFF
+LLVM_CMAKE += -DLLVM_ENABLE_ZLIB=OFF -DLLVM_ENABLE_LIBXML2=OFF -DLLVM_HOST_TRIPLE="$(or $(XC_HOST),$(BUILD_MACHINE))"
```

Comment:
This creates problems on Fedora armv7hl. Indeed, there `BUILD_MACHINE=armv7hl-redhat-linux-gnueabi` by default since that's what `gcc -dumpmachine` returns. Before this PR, LLVM used `armv7l-unknown-linux-gnueabihf` by default instead (as computed by `deps/srccache/llvm-6.0.1/cmake/config.guess`).

I don't really understand why Fedora arm uses `gnueabi` instead of `gnueabihf`, since it uses hard float (which is specified e.g. in `CFLAGS` via `-mfloat-abi=hard`), but [it seems to be a conscious choice](https://src.fedoraproject.org/rpms/gcc/blob/672c273c6d3d662c93934ea3b26b23cecfd4a89f/f/gcc.spec#_258). I couldn't find official GCC/LLVM docs about what these mean. What I can see at least is that LLVM's config.guess script branches on `__ARM_PCS_VFP | $CC_FOR_BUILD -E - 2>/dev/null | grep -q __ARM_PCS_VFP` to decide whether to return `gnueabi` or `gnueabihf`. I guess we should do the same to make this more reliable?